lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -419,12 +419,6 @@ public function commit($entity = null)
         try {
             // Collection deletions (deletions of complete collections)
             foreach ($this->collectionDeletions as $collectionToDelete) {
-                if (! $collectionToDelete instanceof PersistentCollection) {
```

Comment:
This is not possible since `collectionDeletions` is always containing PersistentCollection.
- when looking at the added value they are always persistent collection
- it it was possible to go in this if, we would have called `$collectionToDelete->getMapping()` but `getMapping` only exists for PersistentCollection.

Reply:
This code was deliberately added in https://github.com/doctrine/orm/pull/7861, which is a bit worrying, but also, the test that was added in that PR still passes, so I guess it's fine?

Reply:
Yes. I would say that the developer wanted to be BC
- If not PersistentCollection do like before
- If PersistentCollection add the bugfix
But it's not possible to have something else than PersistentCollection ^^